plugins/edsm.py

```diff
@@ -389,50 +519,58 @@ def worker():
 
 
 # Whether any of the entries should be sent immediately
-def should_send(entries):
-
+def should_send(entries: List[Mapping[str, Any]]) -> bool:
     # batch up burst of Scan events after NavBeaconScan
     if this.navbeaconscan:
         if entries and entries[-1]['event'] == 'Scan':
             this.navbeaconscan -= 1
             if this.navbeaconscan:
                 return False
+
         else:
             assert(False)
-            this.navbeaconscan = 0
+            this.navbeaconscan = 0  # TODO: Unreachable code?
```

Comment:
Need to review the intent of this code and ensure it's actually fulfilled.  Either this is a brainfart and the lines need swapping, or the `assert(False)` was a leftover from development of this code and needs removing.

Reply:
Even if intentional that should `raise ValueError` rather than `assert False`, because as discussed asserts are stripped in -O3, and an assert False without even a message attached does nothing to explain why it exploded

Reply:
I *think* the intent is that the code in the if part of the conditional this is the else of *should* count the Scan events from a NavBeaconScan correctly, decrementing as it finds them.  *That* should always be what reduces this.navbeaconscan from the incoming scan count to zero.  So, if this.navbeaconscan was still true (> 0) but the next event isn't a Scan event, that's an error.  It's assuming the game always stictly follows the NavBeaconScan event with all the Scan events, nothing inbetween I think?

So, yeah, the `assert(False)` was for debugging, to ensure things fell over if the accounting didn't work.  What it should be replaced by is a `logger.error(...)`.  As for the `this.navbeaconscan = 0`, that should be moot as the only way to hit this code is if it was zero anyway (given the value is initialised so is never None, and nowhere is setting it False).

I do wonder about the logic of testing only the current last member of the passed `entries`.  Because if it's more than one Scan event it should be reducing this.navbeaconscan by more than one, and if there are other types of events mixed in it breaks the basic assumption.

The fact we don't have reports of the program crashing all over on Nav Beacon scans probably means the assumptions are correct and that technically `entries[-1]` is just devolving into `entries[0]`.

Reply:
Replaced with a log.error call. Left the `this.navbeaconscan` assignment as its a handy "just in case"